Rubberduck.CodeAnalysis/Inspections/Concrete/ImplicitlyTypedConstInspection.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Resources.Inspections;
+
+namespace Rubberduck.CodeAnalysis.Inspections.Concrete
+{
+    public sealed class ImplicitlyTypedConstInspection : InspectionBase
+    {
+        public ImplicitlyTypedConstInspection(RubberduckParserState state)
+            : base(state) { }
+
+        protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
+        {
+            var declarationFinder = State.DeclarationFinder;
+
+            var implicitlyTypedConsts = declarationFinder.AllDeclarations
+                .Where(declaration => declaration.DeclarationType == DeclarationType.Constant
```

Comment:
As mentioned in a comment already outdated for some reason, this will yield false positives for all built-in constants. (You might want to add a test case in which you load a library, e.g. the VBA library.) You should only look at user declarations here. 

Moreover, you should never get all declarations and then filter for a declaration type. Instead use the members of the `DeclarationFinder` taking a declaration type. Those get the filtered declarations via a dictionary lookup, which is much faster.